sim-administration/uploader/outfile_to_hss_input_converter_test.go

```diff
@@ -0,0 +1,48 @@
+package main
+
+import (
+	"gotest.tools/assert"
+	"testing"
+)
+
+///
+///   The tests
+///
+
+func testKeywordValueParser(t *testing.T) {
+	theMap := make(map[string]string)
+	ParseLineIntoKeyValueMap("ProfileType     : BAR_FOOTEL_STD", theMap)
+
+	assert.Equal(t, "BAR_FOOTEL_STD", theMap["ProfileType"])
+}
+
+func testReadOutputFile(t *testing.T) {
+	sample_output_file_name := "sample_out_file_for_testing.out"
+	record, _ := ReadOutputFile(sample_output_file_name)
+
+	// First parameter to check
+	assert.Equal(t, sample_output_file_name, record.Filename)
+
+	// Check that all the header variables are there
+	assert.Equal(t, record.headerDescription["Customer"], "Footel")
+	assert.Equal(t, record.headerDescription["ProfileType"], "BAR_FOOTEL_STD")
+	assert.Equal(t, record.headerDescription["Order Date"], "2019092901")
+	assert.Equal(t, record.headerDescription["Batch No"], "2019092901")
+	assert.Equal(t, record.headerDescription["Quantity"], "3")
+
+	// Check all the input variables are there
+	assert.Equal(t, record.inputVariables["ICCID"], "8947000000000012141")
+	assert.Equal(t, record.inputVariables["IMSI"], "242017100011213")
+
+	// Check that the output entry set looks legit.
+	assert.Equal(t, 3, len(record.entries))
+	assert.Equal(t, 3, record.noOfEntries)
+}
+
+//
+//  The test suite tying it all together.
+//
+func Test(t *testing.T) {
```

Comment:
Remove this function and export (capitalize) the two others.  `go test` will run all the functions starting with `Test` and accepting a `*testing.T`.

Reply:
before or after record declarations? (after I presume, but just checking).

Reply:
I'm not sure what you mean by record declarations (structs?), but the order of top level declarations has no effect on program behavior.